backend/plan/management/commands/trainrecommender.py

```diff
@@ -374,7 +391,21 @@ def train_recommender(
             if course_data_path is not None
             else courses_data_from_db()
         )
-    course_clusters = generate_course_clusters(courses_data, n_per_cluster)
+
+    if course_descriptions_path is not None:
+        course_descriptions_or_none = courses_data_from_csv(course_descriptions_path)
+    else:
+        course_descriptions_or_none = None
```

Comment:
Could the default here be `dict()`, and if `course_descriptions_path` is not None then set it to `dict(courses_data_from_csv(course_descriptions_path))`? Then we could call this `preloaded_descriptions` or something like that, and in `generate_course_vectors_dict`, we could default to using `get_description` if the course code isn't in the `preloaded_descriptions` dict. Just a suggestion in case we want to relax the requirement that the courses in `course_description_path` have to be a superset of the courses in course_data.

Reply:
Yeah that makes a lot of sense.

The only issue I foresee is that now we're allowing mixing of preloaded descriptions (from the CSV) and database descriptions. This means we remove the invariant that `courses_data` and `course_descriptions` are coupled (ie course descriptions requires course_data, and the courses in the CSVs are expected to match up). I think most of the time when you provide both CSVs you would want them to be coupled, so this opens up some potentially confusing behavior.

With that being said, I think that we can bypass any confusion by providing good documentation.

Reply:
That's fair. If we do implement this, probably fine to leave it undocumented and remove the documentation describing the need for the courses in `course_descriptions` to be a superset of the courses in `courses_data` (maybe just say "descriptions will be taken from `course_descriptions` if specified, and any missing descriptions will be taken from the database, or set to the empty string if not found"). Generally the less need for documentation / explanation the better. Otherwise it's probably fine to keep it the way it is right now, your call.